lib/components/base_component.dart

```diff
@@ -56,13 +56,12 @@ abstract class BaseComponent extends Component {
   @mustCallSuper
   @override
   void render(Canvas canvas) {
+    canvas.save();
     prepareCanvas(canvas);
 
     if (debugMode) {
       renderDebugMode(canvas);
     }
-
-    canvas.save();
     _children.forEach((c) => c.render(canvas));
     canvas.restore();
```

Comment:
Wait, this is wrong... Then the whole preparation of the canvas will be in vain.